components/prism-abap.js

```diff
@@ -0,0 +1,50 @@
+Prism.languages.abap= {
+	'comment': {
+		pattern: /^[\*].*$/gm,
```

Comment:
There is no need to escape the `*` inside a character class.
Also, I think the `$` is useless here, since `.*` will consume everything until the end of line anyway.

Finally, the `g` flag can be removed: it is not used by Prism. (This applies to all your regexps)